Command/PasswordResetRequestHandler.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace SumoCoders\FrameworkMultiUserBundle\Command;
+
+use SumoCoders\FrameworkMultiUserBundle\User\PasswordResetInterface;
+use SumoCoders\FrameworkMultiUserBundle\User\UserInterface;
+use SumoCoders\FrameworkMultiUserBundle\User\UserRepositoryCollection;
+use Swift_Mailer;
+use Swift_Message;
+
+class PasswordResetRequestHandler
+{
+    /**
+     * @var UserRepositoryCollection
+     */
+    private $userRepositoryCollection;
+
+    /**
+     * @var Swift_Mailer
+     */
+    private $mailer;
+
+    /**
+     * @var string
+     */
+    private $bodyText;
+
+    public function __construct(UserRepositoryCollection $userRepositoryCollection, Swift_Mailer $mailer, $bodyText)
+    {
+        $this->userRepositoryCollection = $userRepositoryCollection;
+        $this->mailer = $mailer;
+    }
+
+    /**
+     * Creates a password reset token and sends an email to the user.
+     *
+     * @param PasswordResetRequest $command
+     */
+    public function handle(PasswordResetRequest $command)
+    {
+        $user = $command->getUser();
+        $user->generatePasswordResetToken();
+        $repository = $this->userRepositoryCollection->findRepositoryByClassName(get_class($user));
+        $repository->save($user);
+        $this->sendPasswordResetToken($user);
+    }
+
+    /**
+     * Sends the password reset token to the user.
+     *
+     * @param PasswordResetInterface $user
+     *
+     * @return int
+     */
+    private function sendPasswordResetToken(PasswordResetInterface $user) {
+        $messageBody = $this->getPasswordResetMessage($user);
+
+        $message = Swift_Message::newInstance()
+            ->setSubject('Password reset requested')
+            ->setFrom('send@example.com')
+            ->setTo($user->getEmail())
+            ->setBody( $messageBody, 'text/plain');
+        
+        return $this->mailer->send($message);
```

Comment:
properder zou zijn om een event te dispatchen en in een listener een mail te sturen.